python/ray/train/v2/_internal/execution/context.py

```diff
@@ -86,8 +87,58 @@ class ExecutionContext:
     train_context_callbacks: List["TrainContextCallback"]
 
 
+class TrainContext(ABC):
+    @abstractmethod
+    def get_experiment_name(self) -> str:
+        pass
+
+    @abstractmethod
+    def get_world_size(self) -> int:
+        pass
```

Comment:
This interface should be defined at the user-facing level, rather than the internal context implementation.

Right now we pass through 2 layers of contexts (external + internal), but only the external `ray.train.get_context()` needs to be patched, rather than the internal `get_train_context`.

https://github.com/ray-project/ray/blob/7119e9306c5bedbcfca798fbace9e4e544766f0a/python/ray/train/v2/api/context.py#L60